### ISSUE TITLE
Issues

### ISSUE THREAD

**Issue by oksanaNeil** [Botspot/vdesktop#2]
This is a very good idea. I got it working with a Raspberry pi Buster .img file from one of my
working Raspberry pi computers. I used my normal Raspberry pi 4 with a buster image as the 'host'.
I noted the following:

1.  http pages browse ok. https seem to have a TLS 'unable to get local issuer certificate' problem 
and don't open.

2. Complains of SSH duplicate port 22. Changed this to 12148 in /etc/ssh/sshd_config and
it stopped complaining! https still don't work.

3. It seemed to pick up the same IP as the host?

4. I compiled freeCAD on the guest machine (my first compile) because it is a problem to
get to work on RasPi4, Buster. It errors on trying to run :
  'Program received signal SIGSEGV, Segmentation fault.
  #0  /lib/arm-linux-gnueabihf/libc.so.6(__default_sa_restorer+0) [0xb0c8c120]'

5. Running a 3d printer slicing program called Cura which is an '.AppImage' file it failed to run.

When I closed the guest and flashed the .img file to a SD card and ran the image in my Raspberry pi 4 the image worked fine, new dhcp IP, https ok, FreeCAD and Cura programs worked fine.

I am new to Linux etc but the idea of running experiments in a guest window is really good.

**Comment by Botspot** [comment]
In response to your comments:
1. Before Buster, the guest had a strong internet connection. But once Buster rolled out, the guest usually doesn't detect any internet devices such as Ethernet or WiFi. This is a known bug, but I am unaware of any solutions.
3. That's normal. The guest is configured to "share" the Ethernet connection with the host. So since both OS'es are on the same device, both will appear to have the same IP address.
4. You say FreeCAD doesn't compile on Buster. And you tried to compile it within the guest but it didn't work. Didn't you say that the guest is also Buster so what's the difference?
5. Does that 3d printer program run correctly outside of the guest? It could be due to a lack of OpenGL support in the Xephyr window.

**Comment by Botspot** [comment]
Looks like [FreeCAD isn't working](https://www.raspberrypi.org/forums/viewtopic.php?f=66&t=244151) for others either. So I see no reason why Vdesktop is at fault here.

**Comment by oksanaNeil** [comment]
Hello Botspot,

Thanks for quick reply /comments. Don't be disappointed I like the idea. I started doing things for
pure self satisfaction years ago stick with it !

1. Internet connection. I tried giving it a fixed IP by using it's MAC address if I remember right
    but it did not seem to work. I thought the same IP as the host might have been something to
   do with why I couldn't seem to get certificates right for https. 
   Maybe I will fiddle with NAT or something. Still thinking on this.

2. On the FreeCAD forum a bit later than what you found they got FreeCAD to compile and work
   but a slightly older version and with old python that's not being used in the future.
  This less than ideal solution is good enough for me.  
   I'm looking how to 'capture' this compile and all it' s dependencies into a 'package' so I can
  easily put it back onto my other raspi4 machines. Something like 'virtual machine snapshot'
  or a 'saved sandbox image' or an .AppImage file.
  Here is the forum:
     https://forum.freecadweb.org/viewtopic.php?f=42&t=37458&start=90
 
4.The 3D printer slicer program is in .AppImage format and works ok on a raspi4.
   Your idea about OpenGL maybe something to do with why it does not work in your 'guest'.
  
I don't want to compile it on my raspi4 main machine because I will probable 'trash' it
while messing about !
I am not very good with Linux, I spent 20 years on Windows doing software packaging and
distribution, .msi & AppV packages, for large clients for my sins...

Remember self satisfaction is why you do a job, earning a 'crust' comes second.
